examples/ansible/setup_neuvector.yml

```diff
@@ -0,0 +1,113 @@
+---
+# This Ansible playbook is used to manage the NeuVector workload on a ALP host.
+# The steps are based on : https://build.opensuse.org/package/view_file/SUSE:ALP:Workloads/neuvector-demo/README.md?expand=1
+# and https://documentation.suse.com/alp/micro/html/alp-micro/available-alp-workloads.html#task-run-neuvector-with-podman
+# The playbook supports two operations: setup and removal of NeuVector.
+# The operation is determined by the 'neuvector_state' variable, which can be set to 'present' or 'absent'.
+# When 'neuvector_state' is set to 'present', the playbook will:
+#   - Check if the tools for the NeuVector workload are already installed
+#   - Set SELinux into permissive mode
+#   - Retrieve the NeuVector image
+#   - Execute nevector runlabel INSTALL
+#   - Start the NeuVector service
+#   - Enable the NeuVector service to start on a reboot
+#   - Print a message with the connection details for NeuVector
+# When 'neuvector_state' is set to 'absent', the playbook will:
+#   - Stop the NeuVector service
+#   - Execute nevector runlabel UNINSTALL
+#   - Remove the NeuVector image
+
+- name: Running the NeuVector workload
+  hosts: alphost
+  vars:
+    neuvector_state: present
+    workload:
+      name: neuvector
+      image: registry.opensuse.org/suse/alp/workloads/bci_containerfiles/suse/alp/workloads/neuvector-demo:latest
+
+  tasks:
+    - name: Setup NeuVector
+      when:
+        - neuvector_state == "present"
+      block:
+        - name: Check if workload {{ workload.name }} is already installed
+          become: true
+          containers.podman.podman_container_info:
+            name: "{{ workload.name }}-allinone"
+          register: workload_info
+          changed_when: false
+
+        - name: Set SELinux into permissive mode
+          become: true
+          ansible.posix.selinux:
+            policy: targeted
+            state: permissive
+          when:
+            - workload_info.containers | length == 0
+
+        - name: Retrieve image for workload {{ workload.name }}
+          become: true
+          containers.podman.podman_image:
+            name: "{{ workload.image }}"
+            state: present
+          register: retrieve_workload_image
+          when:
+            - workload_info.containers | length == 0
+
+        - name: Execute nevector runlabel INSTALL
+          become: true
+          ansible.builtin.command: >-
+            podman container runlabel install {{ workload.image }}
+          register: workload_runlabel_install
+          changed_when:
+            - ('already exist' not in workload_runlabel_install.stdout)
+            - workload_info.containers | length == 0
```

Comment:
Isn't the above condition enough?

Reply:
This is still not fixed.

Reply:
Using 'changed_when' to change task status when "already exist" does not appear in the command's output, are you suggesting using "workload_info.containers | length == 0" instead or something else?

Reply:
What happens if you execute `podman container runlabel install {{ workload.image }}` twice? If nothing then the whole `when` can be omitted.

Reply:
The task "Execute nevector runlabel INSTALL" will always run in the block because there's no when condition to prevent it. change_when is only used for reporting "changed" when the command has made any changes. ie, the task will be reported as "changed" only if the string "already exist" does not appear in the stdout of the command.

Reply:
made the suggested changes.^

Reply:
You remove more than needed. Please keep:
```
changed_when:
 - ('already exist' not in workload_runlabel_install.stdout)
```